### PR TITLE
Disable port discovery feature design

### PR DESCRIPTION
**Pull request by rozdy**
Design for disable port discovery feature. Issue #2794

### CODE REVIEW

File: docs/design/network-discovery/port-FSM.puml [hunk 1]
```diff
@@ -39,9 +39,11 @@ state OPERATIONAL {
 
     UP --> DOWN : port-down
     UP : enter / emit port-down into report
-    UP : enter / issue addWatch
+    UP : enter / issue addWatch if discovery enabled
     UP : poll-discovery / emit discovery
     UP : poll-fail / emit fail
+    UP : enable-discovery / issue addWatch
+    UP : disable-discovery / issue delWatch
```
Comment (surabujin) [inline docs/design/network-discovery/port-FSM.puml]: Mention "set enabled = false" here. As far as I understood it is done here.

File: docs/design/network-discovery/port-FSM.puml [hunk 2]
```diff
@@ -39,9 +39,11 @@ state OPERATIONAL {
 
     UP --> DOWN : port-down
     UP : enter / emit port-down into report
-    UP : enter / issue addWatch
+    UP : enter / issue addWatch if discovery enabled
```
Comment (timofei-durakov) [inline docs/design/network-discovery/port-FSM.puml]: UP : enter[discovery-enabled] / issue addWatch

File: docs/design/network-discovery/port-FSM.puml (hunk 1 above)
Comment (rozdy) [reply docs/design/network-discovery/port-FSM.puml]: Added a readme. Port Fsm should read enabled/disabled flag from DB on "UP : enter" so we don't need to save flag manually.

File: docs/design/network-discovery/port-FSM.puml (hunk 2 above)
Comment (rozdy) [reply docs/design/network-discovery/port-FSM.puml]: Fixed.

File: docs/design/network-discovery/disable-port-discovery/domain-model.puml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+   ...
+   __
+   unique constraint on switch_id
+}
+
+class PortProperties {
+   switch : Switch
+   port_no
+   discoveryEnabled
```
Comment (sergii-iakovenko) [inline docs/design/network-discovery/disable-port-discovery/domain-model.puml]: Please, be consistent - either snake_case or camelCase.

File: docs/design/network-discovery/disable-port-discovery/disable-port-discovery.puml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+@startuml
+title Port discovery enable/disable
+
+participant Northbound as NB
+participant nbworker as NBW
+database DB
+control kilda.topo.disco as DI
+box NetworkTopology
+participant SpeakerRouter as SR
```
Comment (sergii-iakovenko) [inline docs/design/network-discovery/disable-port-discovery/disable-port-discovery.puml]: As I understood, SpeakerRouter is for requests that come from Speaker, not NB.

File: docs/design/network-discovery/disable-port-discovery/domain-model.puml (hunk 3 above)
Comment (rozdy) [reply docs/design/network-discovery/disable-port-discovery/domain-model.puml]: Fixed

File: docs/design/network-discovery/disable-port-discovery/disable-port-discovery.puml (hunk 4 above)
Comment (rozdy) [reply docs/design/network-discovery/disable-port-discovery/disable-port-discovery.puml]: For now SpeakerRouter is used for at least two NB messages: IslBfdFlagUpdated and FeatureTogglesUpdate. Probably we need to rework topology or rename SpeakerRouter.

Comment (timofei-durakov) [reply docs/design/network-discovery/disable-port-discovery/disable-port-discovery.puml]: there is a comment on that      // FIXME(surabujin): is it ok to consume this "event" from speaker stream?

Comment (timofei-durakov) [reply docs/design/network-discovery/disable-port-discovery/disable-port-discovery.puml]: P.S. I' refactor that separately

File: docs/design/network-discovery/disable-port-discovery/domain-model.puml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+class PortProperties {
+   switch : Switch
+   port_no
+   discovery_enabled
+   __
+   unique constraint on switch_id+port_no
+}
+
+Switch --> PortProperties
```
Comment (sergii-iakovenko) [inline docs/design/network-discovery/disable-port-discovery/domain-model.puml]: Shouldn't it be opposite direction?

Comment (rozdy) [reply docs/design/network-discovery/disable-port-discovery/domain-model.puml]: I assume relation "Switch -has/owns-> PortProperties" so decided to use this direction. We already have "Switch -has-> SwitchProperties" relation as well.

**Pull request merged**